### PR TITLE
Feature/Initialize the solution

### PR DESCRIPTION
**Pull request by devpro**
Changes:
- Initialize the repository with a README, editorconfig, markdownlint, a minimalist CI pipeline
- Scripts to create a K3s cluster, install Rancher, create a custom downstream cluster from Rancher

Validation:
- Rancher Playground Track

### CODE REVIEW

File: scripts/kubernetes/certificate-management.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#   administrator email address (to receive notifications for Let's Encrypt)
+# Examples:
+#   k8s_create_letsencryptclusterissuer traefik john.wick@google.com
+#######################################
+k8s_create_letsencryptclusterissuer() {
+  local ingressClassname=$1
+  local emailAddress=$2
+
+  echo "Creating certificate issuers using Let's Encrypt..."
+  helm repo add devpro https://devpro.github.io/helm-charts
```
Comment (hierynomus) [inline scripts/kubernetes/certificate-management.sh:37]: This one we might want to move at some point in time

File: scripts/download.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+fatal() {
+  echo '[ERROR] ' "$@" >&2
+  exit 1
+}
+
+verify_system() {
+  info 'Verify system requirements'
+  if [ -x /usr/bin/git ] || type git > /dev/null 2>&1; then
```
Comment (hierynomus) [inline scripts/download.sh:37]: Maybe we should also verify `jq` present as a lot of commands depend on it.

File: scripts/rancher/manager-lifecycle.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  echo "Installing Rancher..."
+  helm repo add rancher-${repository} https://releases.rancher.com/server-charts/${repository}
+  helm repo update
+  helm upgrade --install rancher rancher-${repository}/rancher --namespace cattle-system --create-namespace \
+    --version ${version} \
+    --set replicas=${replicas} \
+    --set hostname=${hostname} \
+    --set ingress.extraAnnotations.'cert-manager\.io/cluster-issuer'=${clusterissuer} \
+    --set ingress.tls.source=secret \
+    --set ingress.tls.secretName=rancher-tls
```
Comment (hierynomus) [inline scripts/rancher/manager-lifecycle.sh:30]: For `v2.9.1` we need to also pass

```
  --set agentTLSMode="system-store"
```

File: scripts/kubernetes/certificate-management.sh (hunk 1 above)
Comment (devpro) [reply scripts/kubernetes/certificate-management.sh:37]: Yes! Added a TODO, really need to do this one (as for the container images)

File: scripts/download.sh (hunk 2 above)
Comment (devpro) [reply scripts/download.sh:37]: Done. Thanks

File: scripts/rancher/manager-lifecycle.sh (hunk 3 above)
Comment (devpro) [reply scripts/rancher/manager-lifecycle.sh]: Added. Thanks!

**Comment by devpro** [comment]
Latest version of the scripts just validated on Rancher playground track

**Pull request merged**